app/models/task.py

```diff
@@ -1,5 +1,41 @@
+from flask import abort, make_response, request
 from app import db
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String(100))
+    description = db.Column(db.String(280))
+    completed_at = db.Column(db.DateTime, nullable=True, default=None)
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'), nullable=True)
+    goal = db.relationship("Goal", back_populates="tasks")
+
+    def validate_task(task_id):
+        '''Validates that task id is valid and exists'''
+        try:
+            task_id = int(task_id)
+        except:
+            abort(make_response({"msg": f"Invalid id: {task_id}"}, 400))
+        
+        task = Task.query.get(task_id)
+        if not task:
+            abort(make_response({"msg": f"Could not find task with id: {task_id}"}, 404))
+        return task
+
+    def is_complete(task):
+        '''Checks whether a task is incomplete or complete.'''
+        if task.completed_at:
+            return True
+        else:
+            return False
```

Comment:
This is set up as a static method, but I recommend making it an instance method.  In a case like this where you have the instance of the class, I suggest writing the function so that you can ask the instance to do the work, rather than reaching inside the instance to do the work for it.
```suggestion
    def is_complete(self):
        '''Checks whether a task is incomplete or complete.'''
        if self.completed_at:
            return True
        else:
            return False
```